source/slang/slang-ir-autodiff-fwd.cpp

```diff
@@ -1578,8 +1587,15 @@ IRInst* ForwardDerivativeTranscriber::transcribe(IRBuilder* builder, IRInst* ori
                     builder->addNameHintDecoration(pair.differential, sb.getUnownedSlice());
                 }
 
-                // Tag the differential inst using a decoration.
-                builder->markInstAsDifferential(pair.differential, as<IRType>(pair.primal->getDataType()));
+                // Tag the differential inst using a decoration (if it doesn't have one)
+                if (!pair.differential->findDecoration<IRDifferentialInstDecoration>() &&
+                    !pair.differential->findDecoration<IRMixedDifferentialInstDecoration>())
+                {
```

Comment:
I don't think it makes sense to decorate types. Why do we need them on types? A type can be used anywhere, including in primal code so marking them as differential/mixed doesn't seem to make sense.

Reply:
Ah, I don't think we decorate types? We decorate the instruction based on the type (this isn't implemented yet though)
but you're right that we should never be tagging types this way.